### PR TITLE
Feat/enhance deploy scripts

### PR DESCRIPTION
**Pull request by whalelephant**
- Splits deploy host and remote chains 
- Save deploys to public networks
- Fix plugin registry instantiate bug 
- Speed up local node tests
- Minor doc updates

### CODE REVIEW

File: cli/utils/fs.ts [hunk 1]
```diff
@@ -12,6 +13,12 @@ export function writeInCacheFolder(fileName: string, content: string, encoding:
     fs.writeFileSync(path.join(cachePath, fileName), content, { encoding });
 }
 
+export function writeToFile(fullPath: string, content: string, encoding: BufferEncoding = "utf8"): void {
+    const dir = path.dirname(fullPath);
+    mkdirp.sync(dir);
```
Comment (j0nl1) [inline cli/utils/fs.ts:18]: Node already include a function to create folder recursively.
https://nodejs.org/api/fs.html#fsrmdirsyncpath-options

**Review by j0nl1 — COMMENTED** [review]
There is a little change needed to do since you are renamed the local-node-script, is also necessary to update the makefile

**Comment by whalelephant** [comment]
> There is a little change needed to do since you are renamed the local-node-script, is also necessary to update the makefile

Hey @j0nl1 I don't think we were using local-node-setup anywhere. We have been testing with the dockers via nodes-setup. Did I miss something?

File: cli/utils/fs.ts (hunk 1 above)
Comment (whalelephant) [reply cli/utils/fs.ts:18]: removed deps. thanks!

**Review by j0nl1 — APPROVED** [review]

**Pull request merged**